helm/chart/maesh/templates/controller/controller-deployment.yaml

```diff
@@ -70,6 +70,9 @@ spec:
             {{- if .Values.mesh }}
             - "--defaultMode={{ .Values.mesh.defaultMode }}"
             {{- end }}
+            {{- if .Values.acl }}
```

Comment:
```suggestion
            {{- if or .Values.acl .Values.smi.enable}}
```

Reply:
Since we are deprecating the `--smi` flag

Reply:
I considered this option but instead we chose to interpret the option and do the backward compatibility directly in the core. The solution you are proposing is compatible only when you're using Maesh via Helm. Most of our users do but iif you do it in the code you can cover Helm installs and manual installs.